euphonic/qpoint_frequencies.py

```diff
@@ -0,0 +1,240 @@
+
+    def get_dispersion(self) -> Spectrum1DCollection:
+        """
+        Creates a set of 1-D bands from mode data
+
+        Bands follow the same q-point order as in the qpts array, with
+        x-axis spacing corresponding to the absolute distances between
+        q-points.  Discontinuities will appear as large jumps on the
+        x-axis.
+
+        Returns
+        -------
+        dispersion
+            A sequence of mode bands with a common x-axis
+        """
+        abscissa = _calc_abscissa(self.crystal.reciprocal_cell(), self.qpts)
+        x_tick_labels = get_qpoint_labels(self.qpts,
+                                          cell=self.crystal.to_spglib_cell())
+        return Spectrum1DCollection(abscissa, self.frequencies.T,
+                                    x_tick_labels=x_tick_labels)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert to a dictionary. See QpointFrequencies.from_dict for
+        details on keys/values
+        """
+        dout = _obj_to_dict(self, ['crystal', 'n_qpts', 'qpts', 'frequencies',
+                                   'weights'])
+        return dout
+
+    def to_json_file(self, filename: str) -> None:
+        """
+        Write to a JSON file. JSON fields are equivalent to
+        from_dict keys
+
+        Parameters
+        ----------
+        filename
+            Name of the JSON file to write to
+        """
+        _obj_to_json_file(self, filename)
+
+    @classmethod
+    def from_dict(cls: T, d: Dict[str, Any]) -> T:
+        """
+        Convert a dictionary to a QpointFrequencies object
+
+        Parameters
+        ----------
+        d
+            A dictionary with the following keys/values:
+
+            - 'crystal': dict, see Crystal.from_dict
+            - 'qpts': (n_qpts, 3) float ndarray
+            - 'frequencies': (n_qpts, n_branches) float ndarray
+            - 'frequencies_unit': str
+
+            There are also the following optional keys:
+
+            - 'weights': (n_qpts,) float ndarray
+        """
+        crystal = Crystal.from_dict(d['crystal'])
+        d = _process_dict(d, quantities=['frequencies'], optional=['weights'])
+        return QpointFrequencies(crystal, d['qpts'], d['frequencies'],
+                                 d['weights'])
+
+    @classmethod
+    def from_json_file(cls: T, filename: str) -> T:
+        """
+        Read from a JSON file. See from_dict for
+        required fields
+
+        Parameters
+        ----------
+        filename
+            The file to read from
+        """
+        return _obj_from_json_file(cls, filename)
+
+    @classmethod
+    def from_castep(cls: T, filename: str) -> T:
+        """
+        Reads precalculated phonon mode data from a CASTEP .phonon file
+
+        Parameters
+        ----------
+        filename
+            The path and name of the .phonon file to read
+        """
+        data = castep._read_phonon_data(filename, read_eigenvectors=False)
```

Comment:
Why is this method private? Isn't it the main purpose of the submodule?

Reply:
Its sort of hinting to users "don't use this, use Euphonic objects please", but since it is being called from elsewhere it should probably be public. Have made it public now.